src/downloader-resolver.ts

```diff
@@ -0,0 +1,10 @@
+import { Downloader } from './Downloader';
+import { GotDownloader } from './GotDownloader';
+
+export async function getDownloaderForSystem(): Promise<Downloader<any>> {
+  // TODO: Resolve the downloader or default to GotDownloader
+  // Current thoughts are a dot-file traversal for something like
+  // ".electron.downloader" which would be a text file with the name of the
+  // npm module to import() and use as the downloader
+  return new GotDownloader();
```

Comment:
Ah! Could you use `await import` here and prevent loading `Got` at all unless we really need to?